extractors/words/difficult_words_extraction/code_snippet_common.md

```diff
@@ -0,0 +1,43 @@
+```python
+import textstat
+from typing import List, Tuple
+import re
+import spacy
+
+def difficult_words_extraction(text: str, extraction_keyword: str, syllable_threshold: int) -> List[Tuple[str, int]]:
+      """
+      @param text: text to extract from
+      @param syllable_threshold: threshold for the number of syllables in a word
+      """
+      nlp = spacy.load("en_core_web_sm")
+      doc = nlp(text)
+
+      difficult_words = textstat.difficult_words_list(text, syllable_threshold)
+
+      text = text.replace(".", " .").replace(",", " ,").replace("!", " !").replace("?", " ?") # add space before punctuation
+
+      difficult_word_positions = []
+      for word in difficult_words: 
+            start, end = re.search(rf"({word})", text).span() # get the position of the word in the text
+            span = doc.char_span(start, end, alignment_mode="expand")
+            difficult_word_positions.append([extraction_keyword, span.start, span.end])
```

Comment:
the brick in the __init__ uses `span.start_char` instead of `span.start` which I also prefer. Maybe this then has to be adapted

Reply:
Changed it!